lib/rugged/commit.rb

```diff
@@ -20,10 +20,24 @@ def header_field?(field)
 
     # Return a diff between this commit and its first parent or another commit or tree.
     #
+    # The commit is treated as the new side of the diff by default
+    #
     # See Rugged::Tree#diff for more details.
     def diff(*args)
-      args.unshift(parents.first) if args.size == 1 && args.first.is_a?(Hash)
-      self.tree.diff(*args)
+      raise ArgumentError("wrong number of arguments (given #{args.length}, expected 0..2") if args.length > 2
+      other, opts = args
+      if other.is_a?(Hash)
+        opts = other
+        other = nil
+      end
+      opts ||= {}
+      # if other is not provided at all (as opposed to explicitly nil, or given)
+      # then diff against the prior commit
+      if args.empty? || args.first.is_a?(Hash)
+        other = parents.first
+        opts[:reverse] = !opts[:reverse] if other
```

Comment:
Why are we checking `if other` here? If `self` is a root commit, `parents.first` will be `nil` but even in that case we want to reverse the order, or it will look the wrong way around.